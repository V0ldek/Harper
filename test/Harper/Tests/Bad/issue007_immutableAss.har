fun :: Integer -> Integer;
fun n = {
    (x :: Integer) = n;
    x += 1;
    return x;
};

main :: Integer;
main = fun 42;