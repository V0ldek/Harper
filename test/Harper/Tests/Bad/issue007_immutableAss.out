Error: cannot assign to an immutable value `x`.
During evaluation of:
x += 1 ;

Located at line 4 column 5

Execution terminated with an error: type error.